addons/point_of_sale/__init__.py

```diff
@@ -5,3 +5,34 @@
 from . import controllers
 from . import report
 from . import wizard
+
+from odoo import api, SUPERUSER_ID
+
+
+def _create_and_assign_picking_types(cr, registry):
+    _create_sequences_and_picking_types(cr, registry)
+    _assign_picking_types(cr, registry)
+
+
+def _create_sequences_and_picking_types(cr, registry):
+    """ Add a PoS picking type on every existing warehouse which don't have one yet.
+    
+    It is necessary if the point_of_sale module is installed after some warehouses were already created.
+    """
+    env = api.Environment(cr, SUPERUSER_ID, {})
```

Comment:
since you have multiple functions in the same parent call (both functions get called), it's better to use an environment manager here to avoid instantiating twice the same env by using the env manager:
```python
with api.Environment.manage():
    env = api.Environment(cr, SUPERUSER_ID, {})
    # do stuff
```
same for `_assign_picking_types`

the first environment you will instantiate will be stored in memory and reused in the second function